pkg/controllerhelpers/handlers.go

```diff
@@ -89,29 +89,41 @@ func (h *Handlers[T]) EnqueueAll(depth int, untypedObj kubeinterfaces.ObjectInte
 	}
 }
 
-func (h *Handlers[QT]) EnqueueOwner(depth int, obj kubeinterfaces.ObjectInterface, operation HandlerOperationType) {
-	controllerRef := metav1.GetControllerOf(obj)
-	if controllerRef == nil {
-		return
-	}
+func (h *Handlers[QT]) EnqueueOwnerIf(precondition func(QT) bool) EnqueueFuncType {
+	return func(depth int, obj kubeinterfaces.ObjectInterface, operation HandlerOperationType) {
```

Comment:
this one could be a named function

Reply:
How? It uses a closure to pass the precondition. Unless you just meant assigning the anonymous func to a variable and returning it?
```
a := func(...) {...}
return a
```

Reply:
nvm, it reads some of the object data as well, so it would have too many args to be a free standing function